core/src/test/scala/unit/kafka/coordinator/group/GroupMetadataTest.scala

```diff
@@ -259,6 +259,47 @@ class GroupMetadataTest {
     assertFalse(group.supportsProtocols(protocolType, Set("range", "foo")))
   }
 
+  @Test
+  def testOffsetRemovalDuringTransitionFromEmptyToNonEmpty(): Unit = {
+    val topic = "foo"
+    val partition = new TopicPartition(topic, 0)
+    val time = new MockTime()
```

Comment:
`group` as defined in the `setUp` method uses `Time.SYSTEM`. It basically means that we are not using the same clock across the unit test. We should probably create a `GroupMetadata` in this test directly. I suppose that this causes the failing tests.

Reply:
makes sense. thanks for catching that.